src/hotspot/share/oops/inlineKlass.cpp

```diff
@@ -243,16 +243,19 @@ Klass* InlineKlass::value_array_klass_or_null() {
 // T_METADATA, drop everything until and including the closing
 // T_VOID) or the compiler point of view (each field of the inline
 // types is an argument: drop all T_METADATA/T_VOID from the list).
+//
+// Inline types could also have fields in abstract super value classes.
+// Use a HierarchicalFieldStream to get them as well.
 int InlineKlass::collect_fields(GrowableArray<SigEntry>* sig, int base_off) {
   int count = 0;
   SigEntry::add_entry(sig, T_METADATA, name(), base_off);
-  for (JavaFieldStream fs(this); !fs.done(); fs.next()) {
```

Comment:
`JavaFieldStream` failed to get the fields in the abstract class. Using `HierarchicalFieldStream` fixes this.